source/postprocess/geoid.cc

```diff
@@ -326,15 +326,25 @@ namespace aspect
       // Get the spherical harmonic coefficients of the density contribution.
       std::pair<std::vector<double>,std::vector<double> > SH_density_coes = density_contribution(outer_radius);
 
-      // Get the spherical harmonic coefficients of the surface and CMB dynamic topography
-      std::pair<std::pair<double, std::pair<std::vector<double>,std::vector<double> > >, std::pair<double, std::pair<std::vector<double>,std::vector<double> > > > SH_dyna_topo_coes;
-      SH_dyna_topo_coes = dynamic_topography_contribution(outer_radius,inner_radius);
-      std::pair<double, std::pair<std::vector<double>,std::vector<double> > > SH_surface_dyna_topo_coes = SH_dyna_topo_coes.first;
-      std::pair<double, std::pair<std::vector<double>,std::vector<double> > > SH_CMB_dyna_topo_coes = SH_dyna_topo_coes.second;
-
-      // Get the density contrast at the surface and CMB
-      const double surface_delta_rho =  SH_surface_dyna_topo_coes.first - density_above;
-      const double CMB_delta_rho = density_below - SH_CMB_dyna_topo_coes.first;
+      std::pair<double, std::pair<std::vector<double>,std::vector<double> > > SH_surface_dyna_topo_coes;
+      std::pair<double, std::pair<std::vector<double>,std::vector<double> > > SH_CMB_dyna_topo_coes;
+      // Initialize the surface and CMB density contrasts with NaNs becasue they may be unused in case of no dynamic topography contribution.
+      double surface_delta_rho = std::numeric_limits<double>::quiet_NaN();
```

Comment:
Please replace this and the line below by:
```suggestion
      double surface_delta_rho = numbers::signaling_nan<double>();
```
Signaling NaNs ensure that this number is not used by crashing if someone tries to use them.